### PR TITLE
Remove forgotten lock_retrieve from 0787397d69

### PR DESCRIPTION
**Pull request by ioguix**
Commit 0787397d69 removed lock_retrieve but I forgot dump_status_file and bin_output on the way.

Reported by Frédéric Yhuel, fix by me.

### CODE REVIEW

File: check_pgactivity [hunk 1]
```diff
@@ -542,6 +542,49 @@ sub find_hostkey($) {
...
+# * close the file and release the lock.
+#
+# The closure is reponsible to define how and where in the structure it wants to
+# record its data.
+#
+# Parameters are :
+#  * $rec_to: a closure responsible to record its data in the given structure ref
+#  * $storage: file to write to
+sub lock_save($$) {
+    my $rec_to   = shift;
```
Comment (rjuju) [inline check_pgactivity:561]: `$rec_to` isn't really self explanatory.  Maybe $fn_record or something like that?

Comment (ioguix) [reply check_pgactivity]: done

File: check_pgactivity [hunk 2]
```diff
@@ -542,6 +542,49 @@ sub find_hostkey($) {
...
+# * write back the edited data to disk
+# * close the file and release the lock.
+#
+# The closure is reponsible to define how and where in the structure it wants to
+# record its data.
+#
+# Parameters are :
+#  * $rec_to: a closure responsible to record its data in the given structure ref
+#  * $storage: file to write to
+sub lock_save($$) {
```
[the remark this answers is not in the document]
Comment (ioguix) [reply check_pgactivity]: done.
* `lock_load` -> `load_all`
* `lock_save` -> `save_internal`

[the remark this answers is not in the document]
Comment (rjuju) [reply check_pgactivity]: 👍

File: check_pgactivity [hunk 3]
```diff
@@ -542,6 +542,50 @@ sub find_hostkey($) {
...
+        eval { $all = retrieve($storage) };
+        die "could not retrieve data from «${storage}»:\n  $@" if $@;
+    }
+
+    $fn_record->($all);
+
+    eval { store( $all, $storage ); };
+    die "could not update data in «${storage}»:\n  $@" if $@;
+
+    # closing the fh removes the lock
```
Comment (rjuju) [inline check_pgactivity:585]: nitpicking (and not this patch fault but still): s/removes/releases/

**Review by rjuju — APPROVED** [review]
LGTM!

If you can update the comment while at it it's great :)

File: check_pgactivity (hunk 3 above)
Comment (ioguix) [reply check_pgactivity]: done and squashed

**Comment by ioguix** [comment]
thanks @rjuju !

**Pull request merged**